dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -893,28 +891,44 @@ public void syncUserMetadataFromGitHub(User user) {
     }
 
     /**
-     * Retrieves a tag from GitHub and creates a version on Dockstore
+     * Retrieves a tag/branch from GitHub and creates a version on Dockstore
      * @param repository Repository path (ex. dockstore/dockstore-ui2)
-     * @param gitReference Tag reference from GitHub (ex. 1.0)
+     * @param gitReference Branch/tag reference from GitHub (ex. refs/tags/1.0)
      * @param workflow Workflow to add version to
      * @param dockstoreYml Dockstore YML sourcefile
      * @return New or updated version
      * @throws IOException
      */
-    public WorkflowVersion createTagVersionForWorkflow(String repository, String gitReference, Workflow workflow, SourceFile dockstoreYml) throws IOException {
+    public WorkflowVersion createVersionForWorkflow(String repository, String gitReference, Workflow workflow, SourceFile dockstoreYml) throws IOException {
         GHRepository ghRepository = getRepository(repository);
-        String refName = "tags/" + gitReference;
-        GHRef ghRef = ghRepository.getRef(refName);
+
+        // Match the github reference (ex. refs/heads/feature/foobar or refs/tags/1.0)
+        Pattern pattern = Pattern.compile("^refs/(tags|heads)/([a-zA-Z0-9]+([./_-]?[a-zA-Z0-9]+)*)$");
+        Matcher matcher = pattern.matcher(gitReference);
+
+        if (!matcher.find()) {
+            String msg = "Reference " + gitReference + " is not of the valid form";
+            LOG.info(msg);
+            throw new CustomWebApplicationException(msg, LAMBDA_FAILURE);
+        }
+        String gitBranchType = matcher.group(1);
+        String gitBranchName = matcher.group(2);
+
+        GHRef ghRef = ghRepository.getRef(gitBranchType + "/" + gitBranchName);
 
         Triple<String, Date, String> ref = getRef(ghRef, ghRepository);
         if (ref == null) {
-            String msg = "Cannot retrieve the workflow reference from GitHub, ensure that " + gitReference + " is a valid tag.";
+            String msg = "Cannot retrieve the workflow reference from GitHub, ensure that " + gitReference + " is a valid branch/tag.";
             LOG.info(msg);
             throw new CustomWebApplicationException(msg, LAMBDA_FAILURE);
         }
 
-        // Find existing version if it exists
-        Optional<WorkflowVersion> existingVersion = workflow.getWorkflowVersions().stream().filter(workflowVersion -> Objects.equals(workflowVersion.getReference(), gitReference)).findFirst();
+        // Delete existing version if it exists
+        Optional<WorkflowVersion> existingVersion = workflow.getWorkflowVersions().stream().filter(workflowVersion -> Objects.equals(workflowVersion.getReference(), gitBranchName)).findFirst();
+        if (existingVersion.isPresent()) {
```

Comment:
This is fine, but if so inclined you could just do `existingVersion.isPresent(v -> workflow.removeWorkflowVersion(v))` instead of needing an if statement an calling `'get` (exact syntax probably wrong)